lib/tasks/multitenant/triggers.rake

```diff
@@ -38,6 +38,84 @@ namespace :multitenant do
     puts "Recreated #{triggers.size} triggers"
   end
 
+  desc 'Fix empty or corrupted tenant_id in accounts'
+  task :fix_corrupted_tenant_id_accounts, %i[batch_size sleep_time] => :environment do |_task, args|
+    batch_size = (args[:batch_size] || 100).to_i
+    sleep_time = (args[:sleep_time] || 1).to_i
+
+    ids = (Rails.application.simple_try_config_for(ENV['FILE']) || [])
+
+    ids.in_groups_of(batch_size).each do |group|
+      puts "Executing update for a batch of size: #{group.size}"
+      group.each do |id|
```

Comment:
An alternative to iterating over the group while delegating more to the DBMS instead:

```ruby
Account.buyers.where(id: group).update_all('tenant_id = provider_account_id')
Account.providers.where(id: group).update_all('tenant_id = id')
```